packages/node/src/integrations/http/SentryHttpInstrumentation.ts

```diff
@@ -316,3 +340,128 @@ function getBreadcrumbData(request: http.ClientRequest): Partial<SanitizedReques
     return {};
   }
 }
+
+/**
+ * This method patches the request object to capture the body.
+ * Instead of actually consuming the streamed body ourselves, which has potential side effects,
+ * we monkey patch `req.on('data')` to intercept the body chunks.
+ * This way, we only read the body if the user also consumes the body, ensuring we do not change any behavior in unexpected ways.
+ */
+function patchRequestToCaptureBody(req: IncomingMessage, normalizedRequest: Request): void {
+  const chunks: Buffer[] = [];
```

Comment:
q: How would this behave for long lived requests?

Reply:
hmm, we only add to this if users use `req.on('data')` themselves. But I guess this _could_ still leave to problems - maybe we add a limit at which we throw data away? 🤔